Library/ViewModels/CommentsViewModel.swift

```diff
@@ -22,15 +35,52 @@ public final class CommentsViewModel: CommentsViewModelType,
   CommentsViewModelInputs,
   CommentsViewModelOutputs {
   public init() {
-    // FIXME: Configure this VM with a project in order to feed the slug in here to fetch comments
-    // Call this again with a cursor to paginate.
-    self.viewDidLoadProperty.signal.switchMap { _ in
-      AppEnvironment.current.apiService
-        .fetchComments(query: comments(withProjectSlug: "bring-back-weekly-world-news"))
-        .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
-        .materialize()
+    let projectOrUpdate = Signal.combineLatest(
+      self.projectAndUpdateProperty.signal.skipNil(),
+      self.viewDidLoadProperty.signal
+    )
+    .map(first)
+    .flatMap { project, update in
+      SignalProducer(value: project.map(Either.left) ?? update.map(Either.right))
+        .skipNil()
     }
-    .observeValues { print($0) }
+
+    let initialProject = projectOrUpdate
+      .flatMap { projectOrUpdate in
+        projectOrUpdate.ifLeft(
+          SignalProducer.init(value:),
+          ifRight: {
+            AppEnvironment.current.apiService.fetchProject(param: .id($0.projectId)).demoteErrors()
+          }
+        )
+      }
+
+    let isCloseToBottom = self.willDisplayRowProperty.signal.skipNil()
+      .map { row, total in row >= total - 3 }
+      .skipRepeats()
+      .filter { isClose in isClose }
+      .ignoreValues()
+
+    let (comments, _, _, _) = paginate(
+      requestFirstPageWith: initialProject,
+      requestNextPageWhen: isCloseToBottom,
+      clearOnNewRequest: false,
+      valuesFromEnvelope: { $0.comments },
+      cursorFromEnvelope: { ($0.slug, $0.cursor ?? "") },
```

Comment:
Do we need to coalesce the cursor? It's optional in the `commentsQuery` function.

Reply:
Oh yea good call!

Reply:
Regarding the `isCloseToBottom` `signal`, are we opposed to just appending a TBVC with a loading spinner as the last cell in the table? (unless all the comments have loaded (ie. total == total comments). That way it's less jarring for the user to have their tableview stop and start immediately.

Reply:
I think let's get this initial functionality in and add that later, we may want to confirm ux with @paulomcg